### DISCUSSION TITLE
docs/basic/public-api/

### DISCUSSION THREAD

**Discussion by giscus** [casdoor/casdoor#3401] [Docs comments]
# docs/basic/public-api/

Casdoor Public API

https://casdoor.org/docs/basic/public-api/

**Comment by Jonny12s** [comment]
Hi. I was woundering about the casdoor API documentation with swagger. I am new to the topic, but was expecting to get a access-token after POSTing https://casdoor.mysite.com/api/login. 

1. ut what shall i put in the Body? After some experimenting i found a object that was working, but the swagger api docs didnt help me there alt all, beause the request body is not specified as in some other api cases.  The response obj is always defined, but less usefull.

working obj: 
```
{
  "application": "app-built-in",
  "organization": "built-in",
  "username": "mytestaccount",
  "autoSignin": true,
  "password": "1234test",
  "signinMethod": "Password",
  "type": "login"
}
```

2. I didnt get a token back to make futher more specified requests. 

Am i missing out on something? How can i find out about request obj structure?

Thanks for your help!

**Reply by hsluoyz** [reply]
@j-menter use OAuth's "Resource Owner Password Credentials Grant" to get access token: https://casdoor.org/docs/how-to-connect/oauth#resource-owner-password-credentials-grant

**Reply by Jonny12s** [reply]
Thanks for you quick Response! 😊 But is it possible to add this to the swagger documentstion or is there a reason for not having it there?  Greetings!
